### PR TITLE
Round instead of Truncate while casting float to decimal

### LINKED ISSUE #2997: I would like to round rather than truncate when casting f64 to decimal
**Is your feature request related to a problem or challenge? Please describe what you are trying to do.**

If I convert `0.06999999` to a decimal with scale 2 I would like to get `0.07` rather than `0.06`

```rust
    #[test]
    #[cfg(not(feature = "force_validate"))]
    fn test_cast_f64_to_decimal128() {
        let decimal_type = DataType::Decimal128(18, 2);
        let array = Float64Array::from(vec![
            Some(0.0699999999),
        ]);
        let array = Arc::new(array) as ArrayRef;
        generate_cast_test_case!(
            &array,
            Decimal128Array,
            &decimal_type,
            vec![
                Some(7_i128),
            ]
        );
    }
```

This fails with:

```
Left:  6
Right: 7
```

**Describe the solution you'd like**

Provide an option for round vs truncate?
Default to round?

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by waitingkuo**
# Which issue does this PR close?

Closes #2997

# Rationale for this change
 

casting 0.06999999 to `Decimal(18, 2)` should be  0.07 instead of 0.06

# What changes are included in this PR?

add `.round()` before casting to int

# Are there any user-facing changes?

yes, we round the floating number instead of truncating it

### CODE REVIEW

File: arrow/src/compute/kernels/cast.rs [hunk 1]
```diff
@@ -3192,8 +3194,8 @@ mod tests {
             Some(2.2),
             Some(4.4),
             None,
-            Some(1.123_456_7),
-            Some(1.123_456_7),
+            Some(1.123_456_4), // round down
+            Some(1.123_456_7), // round up
```
Comment (waitingkuo) [inline arrow/src/compute/kernels/cast.rs:3198]: i change the test cases to 
1. round down
2. round up

File: arrow/src/compute/kernels/cast.rs [hunk 2]
```diff
@@ -297,6 +297,9 @@ pub fn can_cast_types(from_type: &DataType, to_type: &DataType) -> bool {
 /// * Time32 and Time64: precision lost when going to higher interval
 /// * Timestamp and Date{32|64}: precision lost when going to higher interval
 /// * Temporal to/from backing primitive: zero-copy with data type change
+/// * Casting from `float32/float64` to `Decimal(precision, scale)` rounds to the `scale` decimals 
+///   (i.e. casting 6.4999 to Decimal(10, 1) becomes 6.5). This is the breaking change from `26.0.0`. 
+///   It used to truncate it instead of round (i.e. outputs 6.4 instead)
```
Comment (waitingkuo) [inline arrow/src/compute/kernels/cast.rs:302]: @Jimexist i updated the doc here

File: arrow/src/compute/kernels/cast.rs [hunk 3]
```diff
@@ -5994,4 +6003,50 @@ mod tests {
...
+    #[cfg(not(feature = "force_validate"))]
+    fn test_cast_f64_to_decimal128() {
+        // to reproduce https://github.com/apache/arrow-rs/issues/2997
+
+        let decimal_type = DataType::Decimal128(18, 2);
+        let array = Float64Array::from(vec![
+            Some(0.0699999999),
+            Some(0.0659999999),
+            Some(0.0650000000),
+            Some(0.0649999999),
```
Comment (waitingkuo) [inline arrow/src/compute/kernels/cast.rs:6017]: @Jimexist added the 0.065 case which will be rounded up to 0.07

**Comment by waitingkuo** [comment]
thank you @Jimexist

**Review by tustvold — APPROVED** [review]

**Pull request merged**